### ISSUE TITLE
zone2sql may crash with invalid zone file as input

### ISSUE THREAD

**Issue by zeha** [PowerDNS/pdns#2126]
This zone file causes zone2sql to crash in a lexical_cast:

```
powerdns-broken.   86400   IN      SOA (   powerdnssec1.ds9a.nl. ahu.ds9a.nl. 1343746984 10800 3600 604800 10800
powerdns-broken.com.   3600    IN  NS  powerdnssec2.ds9a.nl.
```

```
Core was generated by `../pdns/zone2sql --gmysql --zone=crashes/id:000000,sig:11,src:000000,op:flip1,p'.
Program terminated with signal SIGSEGV, Segmentation fault.
#0  lexical_cast_impl (arg=@0x1f0: <error reading variable>) at /usr/include/boost/lexical_cast.hpp:2374
2374	                if(!(interpreter.operator <<(arg) && interpreter.operator >>(result)))
(gdb) bt
#0  lexical_cast_impl (arg=@0x1f0: <error reading variable>) at /usr/include/boost/lexical_cast.hpp:2374
#1  lexical_cast<std::basic_string<char>, int> (arg=@0x1f0: <error reading variable>) at /usr/include/boost/lexical_cast.hpp:2543
#2  ZoneParserTNG::getLineOfFile (this=this@entry=0x7fffe0171900) at zoneparser-tng.cc:243
#3  0x00007f996dca7205 in ZoneParserTNG::makeTTLFromZone (this=this@entry=0x7fffe0171900, str="powerdns-broken.com.") at zoneparser-tng.cc:122
#4  0x00007f996dcd4974 in ZoneParserTNG::get (this=0x7fffe0171900, rr=..., comment=<optimized out>) at zoneparser-tng.cc:441
#5  0x00007f996d9bc4fe in main (argc=3, argv=<optimized out>) at zone2sql.cc:436
#6  0x00007f996c9c0b45 in __libc_start_main (main=0x7f996d9b8b50 <main>, argc=3, argv=0x7fffe0171b18, init=<optimized out>, fini=<optimized out>, rtld_fini=<optimized out>, 
    stack_end=0x7fffe0171b08) at libc-start.c:287
#7  0x00007f996d9c424f in _start ()
```

**Comment by cmouse** [comment]
That's out of zone data right?

**Comment by zeha** [comment]
out of zone is probably not the cause here.

**Comment by pieterlexis** [comment]
The issue with a broken SOA line is fixed in master. This was fixed by #3271.
